src/ShowingTransitionSupport.js

```diff
@@ -0,0 +1,221 @@
+		return function () {
+			sup.apply(this, arguments);
+			this.showingDuration = (this.showingDuration === undefined) ? null      : this.showingDuration;
+			this.hidingDuration  = (this.hidingDuration  === undefined) ? null      : this.hidingDuration;
+			this.shownMethod     = (this.shownMethod     === undefined) ? null      : this.shownMethod;
+			this.hiddenMethod    = (this.hiddenMethod    === undefined) ? null      : this.hiddenMethod;
+			this.shownClass      = (this.shownClass      === undefined) ? 'shown'   : this.shownClass;
+			this.hiddenClass     = (this.hiddenClass     === undefined) ? 'hidden'  : this.hiddenClass;
+			this.hidingClass     = (this.hidingClass     === undefined) ? 'hiding'  : this.hidingClass;
+			this.showingClass    = (this.showingClass    === undefined) ? 'showing' : this.showingClass;
+			this.showingChanged();
+		};
+	}),
+
+	/**
+	* Overrides the showingChanged handler to add support for transitions at the right times and
+	* places.
+	*
+	* @method
+	* @private
+	*/
+	showingChanged: kind.inherit(function (sup) {
+		return function (sender, ev) {
+			var args = arguments;
+
+			// Prepare our visual state
+			this.applyStyle('display', null);
+			this.applyStyle('visibility', null);
+			if (this.showing) {
+				// Reset our state classes, in case we switched mid-stream
+				this.removeClass(this.hidingClass);
+				this.removeClass(this.hiddenClass);
+				sup.apply(this, args);
+				if (this.showingDuration && this.generated) {
+					this.set('showingTransitioning', true);
+					// Start transition: Apply a class and start a timer.
+					// When timer finishes, run the exit function,
+					// remove the transitioning class
+					// and add the final-state class
+					this.addClass(this.showingClass);
+					this.startJob('showingTransition', function () {
+						utils.call(this, this.shownMethod);	// Run the supplied method.
+						this.removeClass(this.showingClass);
+						this.addClass(this.shownClass);
+						this.set('showingTransitioning', false);
+					}, this.showingDuration);
+				} else {
+					// No transition, just a shown class.
+					this.addClass(this.shownClass);
+				}
+			} else {
+				// Reset our state classes, in case we switched mid-stream
+				this.removeClass(this.showingClass);
+				this.removeClass(this.shownClass);
+				if (this.hidingDuration && this.generated) {
+					this.set('showingTransitioning', true);
+					this.addClass(this.hidingClass);
+					this.startJob('showingTransition', function () {
+						utils.call(this, this.hiddenMethod);	// Run the supplied method.
+						this.removeClass(this.hidingClass);
+						this.addClass(this.hiddenClass);
+						this.set('showingTransitioning', false);
+						sup.apply(this, args);
+						this.applyStyle('visibility', 'hidden');
+						this.applyStyle('display', null);
+					}, this.hidingDuration);
+				} else {
+					// No transition, just a hidden class.
+					this.removeClass(this.hidingClass);
```

Comment:
It's necessary as back-up, in case the transition is cut short in the middle (for the edge case of removing the `hidingDuration` during the transition itself), it clears out the previous class so "hiding" and "hidden" don't exist on the node at the same time.

It's also very low cost. Reading a property and doing an string.index on it.